src/pynwb/__init__.py

```diff
@@ -349,6 +349,32 @@ def export(self, **kwargs):
         kwargs['container'] = nwbfile
         super().export(**kwargs)
 
+    @docval({'name': 'neurodata_type', 'type': str, 'doc': 'The neurodata type to search for.'},
+            {'name': 'namespace', 'type': str, 'doc': 'The namespace of the neurodata type to search for.'},)
+    def get_neurodata_type_objs(self, neurodata_type: str, namespace: str):
+        """Return all PyNWB objects in the file that have a neurodata type from a namespace.
+
+        This works regardless of whether the extension was imported earlier in the python execution.
+
+        All objects that are instances of the class associated with the given neurodata_type in the
+        given namespace will be returned. This includes objects that are instances of a subclass.
+
+        For example, if an extension defines a new neurodata type OnePhotonSeries that extends
+        ImageSeries, then `get_neurodata_type_objects_in_file(io, "ImageSeries", "core")` will
+        include all OnePhotonSeries objects.
+
+        .. code-block:: python
+
+            from pynwb import NWBHDF5IO
+            with NWBHDF5IO(filepath, mode="r", load_namespaces=True) as io:
+                obj_list = io.get_neurodata_type_objs("Subject", "core")
+
+        """
+        pynwb_cls = self.manager.type_map.get_dt_container_cls(neurodata_type, namespace)
+
+        read_nwbfile = self.read()
+        return read_nwbfile.find_all_of_class(pynwb_cls)
```

Comment:
Could you clarify what use-case this function addresses that `NWBFile.find_all_of_class` does not?

Reply:
I added a comment in the docstring. It says: This method is useful for getting neurodata type objects from cached extensions where you do not have easy access to a python class to pass to `NWBFile.find_all_of_class`.

Reply:
Makes sense. But it seems a bit strange that this functionality is in two different places. Also, with this we would need to duplicate this also for Zarr. Instead, I'm wondering whether `find_all_of_class` can be modified to do both, something like:

```
@docval({'name': 'neurodata_types', 'type': (type, str), 'doc': 'The PyNWB container class to search for instances of or the string with the name of the neurodata_type'},)
def find_all(self, neurodata_type):
     if isinstance(neurodata_type, str):
         pynwb_cls = self.io.manager.type_map.get_dt_container_cls(neurodata_type, namespace)
     else:
        pynwb_cls = neurodata_type
     ret = [obj for obj in self.objects.values() if isinstance(obj, pynwb_class)]
     return ret
```

Reply:
Yeah, that works too. I'll update it